### ISSUE TITLE
Is dask.dataframe.sample with replacement?

### ISSUE THREAD

**Issue by terrytangyuan** [blaze/dask#892]
Hi dask team, 

Is this method random sampling with replacement or without replacement? http://dask.pydata.org/en/latest/dataframe-api.html#dask.dataframe.core.DataFrame.sample

Thanks,
Yuan

**Comment by mrocklin** [comment]
At the moment it's without replacement.  I've just added sampling with replacement in https://github.com/blaze/dask/pull/893

**Comment by terrytangyuan** [comment]
Thanks!

**Comment by terrytangyuan** [comment]
@mrocklin Is there a way to output the sampled index instead of the actual value?

**Comment by mrocklin** [comment]
I'm not sure I understand.  Can you provide a small example?

**Comment by terrytangyuan** [comment]
So for example for an array, arr = ["a", "b", "c"], arr.sample(1/3) gives 1, 2, or 3, instead of "a", "b", or "c"

**Comment by mrocklin** [comment]
Sadly no, there is no simple way to do this.  Perhaps you can store an incrementing index column along with your data?

**Comment by terrytangyuan** [comment]
Yeah ok. Thanks though.

**Comment by hcarlens** [comment]
@mrocklin is there a way to repeatedly sample from disk into ram without replacement even between sampling operations? 

I expect this isn't possible at the moment given there isn't a way to signify "end of iteration" when we've sampled all the data. 

I'm looking to use this to iterate over a larger-than-RAM training set in a shuffled way, with different shuffling between epochs.
